apis/pipelines/v1alpha5/run_conversion.go

```diff
@@ -1,55 +1,33 @@
 package v1alpha5
 
 import (
+	"github.com/sky-uk/kfp-operator/apis/pipelines"
 	hub "github.com/sky-uk/kfp-operator/apis/pipelines/v1alpha6"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
 func (src *Run) ConvertTo(dstRaw conversion.Hub) error {
 	dst := dstRaw.(*hub.Run)
 
-	dst.ObjectMeta = src.ObjectMeta
+	err := pipelines.TransformInto(src, &dst)
+	if err != nil {
+		return err
+	}
+
 	dst.Spec.Provider = getProviderAnnotation(src)
 	removeProviderAnnotation(dst)
-	dst.Spec.Pipeline = hub.PipelineIdentifier{
-		Name:    src.Spec.Pipeline.Name,
-		Version: src.Spec.Pipeline.Version,
-	}
-	dst.Spec.ExperimentName = src.Spec.ExperimentName
-	dst.Spec.RuntimeParameters = convertRuntimeParametersTo(src.Spec.RuntimeParameters)
-	dst.Spec.Artifacts = convertArtifactsTo(src.Spec.Artifacts)
-	dst.Status.ProviderId.Provider = src.Status.ProviderId.Provider
-	dst.Status.ProviderId.Id = src.Status.ProviderId.Id
-	dst.Status.SynchronizationState = src.Status.SynchronizationState
-	dst.Status.Version = src.Status.Version
-	dst.Status.ObservedGeneration = src.Status.ObservedGeneration
-	dst.Status.ObservedPipelineVersion = src.Status.ObservedPipelineVersion
-	dst.Status.CompletionState = hub.CompletionState(src.Status.CompletionState)
-	dst.Status.MarkedCompletedAt = src.Status.MarkedCompletedAt
-	dst.Status.Conditions = hub.Conditions(src.Status.Conditions)
+
 	return nil
 }
 
 func (dst *Run) ConvertFrom(srcRaw conversion.Hub) error {
 	src := srcRaw.(*hub.Run)
 
-	dst.ObjectMeta = src.ObjectMeta
-	setProviderAnnotation(src.Spec.Provider, &dst.ObjectMeta)
-	dst.Spec.Pipeline = PipelineIdentifier{
-		Name:    src.Spec.Pipeline.Name,
-		Version: src.Spec.Pipeline.Version,
+	err := pipelines.TransformInto(src, &dst)
+	if err != nil {
+		return err
 	}
-	dst.Spec.ExperimentName = src.Spec.ExperimentName
-	dst.Spec.RuntimeParameters = convertRuntimeParametersFrom(src.Spec.RuntimeParameters)
```

Comment:
I haven't checked each function, but for example this one is located in `conversion.go` rather than `run_conversion.go` because it was shared with `runconfiguration_conversion.go`. Now that it's no longer used here, I think it should live in `runconfiguration_conversion.go`, which is similar to how it's done in other places.